### PR TITLE
docs(store): add breaking change with mock store teardown for jasmine tests

### PR DESCRIPTION
**Pull request by daniel-sc**
see #3264

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/migration/v13.md [hunk 1]
```diff
@@ -97,6 +97,17 @@ StoreModule.forFeature(
...
 The `initialState` provided via the configuration object is now typed and needs to match the interface of the state interface.
 
+#### Testing: Reset mock store
+
+Mock stores are not reset automagically after each test. 
+
+When using jasmine, you can reset the mock store after each test by adding the following to your `test.ts`:
+
+```ts
+jasmine.getEnv().afterEach(() => {
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/migration/v13.md:107]: Add the import here also

```suggestion
import { MockStore } from '@ngrx/store/testing';

jasmine.getEnv().afterEach(() => {
```

File: projects/ngrx.io/content/guide/migration/v13.md [hunk 2]
```diff
@@ -97,6 +97,17 @@ StoreModule.forFeature(
 
 The `initialState` provided via the configuration object is now typed and needs to match the interface of the state interface.
 
+#### Testing: Reset mock store
+
+Mock stores are not reset automagically after each test. 
+
+When using jasmine, you can reset the mock store after each test by adding the following to your `test.ts`:
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/migration/v13.md:104]:
```suggestion
When using Jasmine, reset the mock store after each test by adding the following to the `test.ts`:
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by timdeschryver — CHANGES_REQUESTED** [review]
I'm rejecting this to prevent a merge.
I want to try a workaround (I got an idea based on this PR) locally so this wouldn't be an issue anymore.

File: projects/ngrx.io/content/guide/migration/v13.md [hunk 3]
```diff
@@ -97,6 +97,19 @@ StoreModule.forFeature(
...
+
+Mock stores are not reset automagically after each test. 
+
+When using Jasmine, reset the mock store after each test by adding the following to the `test.ts`:
+
+```ts
+import { MockStore } from '@ngrx/store/testing';
+
+jasmine.getEnv().afterEach(() => {
+  TestBed.inject(MockStore, null)?.resetSelectors();
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/migration/v13.md:110]:
```suggestion
  TestBed.inject(MockStore)?.resetSelectors();
```
```suggestion
  TestBed.inject(MockStore, null)?.resetSelectors();
```

File: projects/ngrx.io/content/guide/migration/v13.md [hunk 4]
```diff
@@ -97,6 +97,19 @@ StoreModule.forFeature(
 
 The `initialState` provided via the configuration object is now typed and needs to match the interface of the state interface.
 
+#### Testing: Reset mock store
+
+Mock stores are not reset automagically after each test.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/migration/v13.md:102]: I think adding a "why note" can help to clear things up.


```suggestion
Mock stores are not reset automagically after each test. 

Note that this only applicable when the Angular `TestBed` isn't teared down by Angular, for more info see the [ModuleTeardownOptions options](https://angular.io/api/core/testing/ModuleTeardownOptions).
```

File: projects/ngrx.io/content/guide/migration/v13.md [hunk 5]
```diff
@@ -97,6 +97,19 @@ StoreModule.forFeature(
...
+
+When using Jasmine, reset the mock store after each test by adding the following to the `test.ts`:
+
+```ts
+import { MockStore } from '@ngrx/store/testing';
+
+jasmine.getEnv().afterEach(() => {
+  TestBed.inject(MockStore, null)?.resetSelectors();
+});
+```
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/migration/v13.md:112]: Let's also add a jest example:

```suggestion
```ts
import { MockStore } from '@ngrx/store/testing';

jasmine.getEnv().afterEach(() => {
  TestBed.inject(MockStore, null)?.resetSelectors();
});
```

When using Jest, reset the mock store after each test by using a `afterEach` hook:

```ts
import { MockStore } from '@ngrx/store/testing';

afterEach(() => {
  TestBed.inject(MockStore)?.resetSelectors();
});
```
```

Comment (timdeschryver) [reply projects/ngrx.io/content/guide/migration/v13.md:112]: The suggestions are action weird, the following can be appended:

```ts
import { MockStore } from '@ngrx/store/testing';

afterEach(() => {
  TestBed.inject(MockStore)?.resetSelectors();
});
```

**Review by timdeschryver — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks @daniel-sc!

**Pull request merged**